app/client/src/modules/hash.test.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+let chai = require('chai'),
+    hash = require('./hash');
+
+describe('hash.js', () => {
+
+    let items = [{
+        id: 'item1',
+        value: 1
+    }, {
+        id: 'item2',
+        value: 2
+    }, {
+        id: 'item3',
+        value: 3
+    }];
+
+    it('creates hash from array', () => {
+
+        let hashed = hash(items, 'id');
+
+        chai.assert.equal(1, hashed.item1.value);
+        chai.assert.equal(2, hashed.item2.value);
+        chai.assert.equal(3, hashed.item3.value);
+
+        chai.assert.isUndefined(hashed.item4);
+
+    });
+});
```

Comment:
Keep in mind that lodash has solved every collection problem ever. You can do a  `var hash = require('lodash/collection/indexBy')` and browserify will only include that used module instead of all of lodash. 

https://lodash.com/docs#indexBy